### DISCUSSION TITLE
curl -sS Docker effect

### DISCUSSION THREAD

**Discussion by 99hops** [MacPaw/symfony-health-check-bundle#58] [General]
I have a general question about this `CMD curl -sS {{your host}}/health || exit 1`

For example `return new Response(self::CHECK_RESULT_NAME, false, 'Entity Manager Not Found.');` has no different effect from `return new Response(self::CHECK_RESULT_NAME, true, 'ok');` to `curl -sS` as they all return the same `return new JsonResponse($resultHealthCheck, Response::HTTP_OK);`.

How is this going to set the Docker container to unhealthy?

**Comment by Yozhef** [comment]
@nforced thank you for your request - we will study this issue and provide an answer.

**Reply by 99hops** [reply]
Thanks, in the meanwhile I found a working solution `curl -fks https://localhost/health || exit 1` but for this to work http status code returned has to be > 400 like 503 for example. So curl needs -f to fail on http code > 400.

**Comment by IiiigorGG** [comment]
I think this will be a great improvement, thank you for noticing. We've decided to include this feature in next release according to [health check pattern](https://swsmile.info/post/design-pattern-microservices-health-check-pattern/) so that `/health` and `/ping` endpoints return error http status code (500 for example) if checks fail.
